### ISSUE TITLE
frame dimension is 0 in AMBER netcdf files written using parmed NETCDFReporter

### ISSUE THREAD

**Issue by nrkssa** [ParmEd/ParmEd#1217]
I am using OpenMM to run implicit solvent MD simulations using the AMBER force field. The trajectories are written as AMBER netcdf files using the `parmed netCDFreporter`. The relevant part of the script is as below

```
from parmed.openmm import NetCDFReporter
netcdfreporter = NetCDFReporter('prod.nc', 1000, crds=True)
.
.
simulation.reporters.append(netcdfreporter)
```

The size of the resulting netcdf file is around 8 GB for nearly 30000 frames. However, I could not open the netcdf file in MDtraj/cpptraj. `cpptraj` fails with an error message `"frame dimension is 0"`,   while `mdtraj` reports `0 frames`.

```
md.load_netcdf('prod.nc', top='initial.parm7')
<mdtraj.Trajectory with 0 frames, 20504 atoms, 1338 residues, without unitcells at 0x7fe0cfa05820>
```

The attached zip file contains the OpenMM script, AMBER7 topology file,  output of dataReporter, and the resulting netcdf file with 14 frames. It would be very helpful if you can look into this and help me figure out the issue. 

The package versions are as below.
```
python: 3.9.2
openmm: 7.5.0.dev-b49b82e 
parmed: 3.4.1
mdtraj: 1.9.5
cpptraj: 5.1.0
```

Here is the info on the failing netcdf file:
```
$ncinfo  prod.nc
<class 'netCDF4._netCDF4.Dataset'>
root group (NETCDF3_64BIT_OFFSET data model, file format NETCDF3):
    Conventions: AMBER
    ConventionVersion: 1.0
    application: AmberTools
    program: ParmEd
    programVersion: 3.4.0+11.g1be8ca0f
    title: ParmEd-created trajectory
    dimensions(sizes): frame(0), spatial(3), atom(20504)
    variables(dimensions): |S1 spatial(spatial), float32 time(frame), float32 coordinates(frame, atom, spatial)
    groups:
```

Thanks in advance

[test_trajectory.zip](https://github.com/ParmEd/ParmEd/files/7988562/test_trajectory.zip)

Best
Ram

**Comment by swails** [comment]
Was this using the built-in NetCDF library or was it using the netCDF4 library that accelerates writing?

**Comment by swails** [comment]
I'll see if I can reproduce this.  I wonder if there was some kind of finalization that was missing.

Can you share the full script you used?

**Comment by nrkssa** [comment]
@swails  parmed and openmm were both installed via conda, which means it must most probably be using the built-in library.  The full script including the start up files are in the zip archive I have uploaded in my original report.

**Comment by swails** [comment]
Is `netCDF4` installed in the environment?

**Comment by nrkssa** [comment]
yes, nc-config --version in the environment shows that version 4.6.1 is installed.

**Comment by nrkssa** [comment]
however, netCDF4 module is not installed in the env

**Comment by swails** [comment]
Thanks, I will incorporate a fix for this issue and a workaround to read these incomplete trajectory files.  Thanks for reporting the fix!

**Comment by nrkssa** [comment]
Thanks @swails
